tests/network-tests/src/Api.ts

```diff
@@ -888,18 +890,20 @@ export class Api {
   }
 
   // Looks for the first occurance of an expected event, and resolves.
-  // Use this when the event we are expecting is not particular to a specific extrinsic
-  public waitForSystemEvent(eventName: string): Promise<Event> {
-    return new Promise(async (resolve) => {
-      const unsubscribe = await this.api.query.system.events<Vec<EventRecord>>((events) => {
-        events.forEach((record) => {
-          if (record.event.method && record.event.method.toString() === eventName) {
-            unsubscribe()
-            resolve(record.event)
-          }
-        })
+  // Use this when the event we are expecting is not particular to a specific extrinsic,
+  // Normally events emitted from on_initialize() or on_finalize() in a call
+  public async waitForSystemEvent(eventName: string): Promise<InvertedPromise<Event>> {
```

Comment:
This name also may now be a little misleading

Reply:
agree

Reply:
will drop this, it is not used anywhere